cocos/2d/CCMenu.cpp

```diff
@@ -273,7 +274,11 @@ bool Menu::onTouchBegan(Touch* touch, Event* event)
         _state = Menu::State::TRACKING_TOUCH;
         _selectedWithCamera = camera;
         _selectedItem->selected();
-        
+        if (_eventType == EventType::TOUCH_DOWN) 
+        {
+            _selectedItem->activate();
+        }
+
```

Comment:
Because if event_type was `TOUCH_UP`,  `activate` should not be called when touch is began.